specification/privatedns/resource-manager/readme.md

```diff
@@ -183,6 +211,10 @@ directive:
     from: privatedns.json
     suppress: EnumInsteadOfBoolean
     reason: Booleans are used to indicate binary states of the property, enum is not appropriate.
+  - where: $.definitions.VirtualNetworkLinkProperties.properties.nxdomainRedirect
+    from: privatedns.json
+    suppress: EnumInsteadOfBoolean
+    reason: Booleans are used to indicate binary states of the property, enum is not appropriate.
```

Comment:
>reason: Booleans are used to indicate binary states of the property, enum is not appropriate. [](http://example.com/codeflow?start=4&length=93)

I'm not getting why enums are an inappropriate representation of binary states. Choice of a binary state is an API design decision and it often makes sense to avoid so APIs don't gradually turn into big boolean vectors. But even when the decision is taken that a property should be present that can only ever have 2 values, I'm not seeing how boolean is superior to enum. Using an enum allows you to use better names for the two values, and you can add descriptions for each value.

In this case, something like privateLinkResolutionPolicy: { FailOnNonexistentDomain, FallbackToPublic } makes sense to me. For a client, it's easier to consume/understand/remember than nxDomainRedirect: { True, False }

Can you explain this objection/suppression better?

Reply:
Thanks @mentat9 for the suggestion. I agree with your point, and the backend teams have internally discussed and agreed to switch to enum instead. 
I'll be updating this pull request shortly. Do I also need to update the PR on the private repo and seek arm approval again for it? (I'm afraid it might take 3-4 days to get through the review queue and have that signed off).

Reply:
Cool, thanks @pranavpnair. No need to go back and rev the private spec as long as it is for the same API version.

Reply:
@mentat9 I've updated the private repo with the suggested changes.

https://github.com/Azure/azure-rest-api-specs-pr/pull/18530